### PR TITLE
chore: use `github-slugger` for markdown anchors

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Making use of slugify function from old site

reference:
- https://github.com/eslint/eslint/pull/16066#issuecomment-1168689803

#### Is there anything you'd like reviewers to focus on?

There is one another slugger library 

- https://github.com/eslint/eslint/blob/b6aee9591ecc2e2f5738ab8bef20faac1e05b5c3/docs/package.json#L42

that we add as filter in eleventy and have used it inside only one file 
- https://github.com/eslint/eslint/blob/b6aee9591ecc2e2f5738ab8bef20faac1e05b5c3/docs/src/pages/rules.md?plain=1#L26

The problem here is that this library `Coerces foreign symbols to their English equivalent` [link](https://github.com/simov/slugify/blob/master/config/charmap.json) for more details. so the `problems & statements` is turned into `problems-and-statements` and added inside id. How ever I don't see this id being referenced anywhere. If it is not referenced anywhere (please help confirm this) we can go ahead and remove `slugify` library. 😄

### CODE REVIEW

**Comment by mdjermanovic** [comment]
> How ever I don't see this id being referenced anywhere.

I don't think these sections are referenced from other pages, and even if they are, it's more likely that those links are still using IDs from the old website.

**Comment by mdjermanovic** [comment]
I think for consistency it would be good that IDs are always generated the same way, so this should be `id="layout-formatting"`.

**Comment by harish-sethuraman** [comment]
So I tried to replace with `github-slugger` instead of `slugify` but seems like `github-slugger` produces `layout--formatting` with that same implementation as `archive-website`. Seems like the older site didn't use that (github-slugger) implementation as well. It was using `anchor-js` so we might want to replace `--` with `-` in the new docs site while adding anchors (through slugify filters). Thoughts?

- https://github.com/eslint/archive-website/blob/29f448bb399fbeb94446afde7e5d74ca1b46348a/src/js/main/index.js#L35-L37

**Comment by mdjermanovic** [comment]
If there's already `id` on the element, `anchor-js` uses that `id`, so most `id`s were generated by `github-slugger`. But these particular headers on the old rules page apparently didn't have `id`s, so `anchor-js` assigned them `id`s per its own slugger. Since that's the case, I think it isn't important to retain the exact same `id`s, and `layout--formatting` looks good to me if that's what `github-slugger` produces.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

I'll leave this open in case someone else wants to review it before merging.

File: docs/.eleventy.js [hunk 1]
```diff
@@ -154,7 +161,13 @@ module.exports = function(eleventyConfig) {
 
     eleventyConfig.setLibrary("md",
         markdownIt({ html: true, linkify: true, typographer: true })
-            .use(markdownItAnchor, {})
+            .use(markdownItAnchor, {
+                slugify(text) {
```
Comment (nzakas) [inline docs/.eleventy.js:165]: We can shorten this to just `{ slugify }` using shorthand notation. (Sorry, can’t do multi line suggestions on mobile)

Comment (harish-sethuraman) [reply docs/.eleventy.js]: NP :) I've updated the PR thanks.

**Review by mdjermanovic — APPROVED** [review]

**Review by nzakas — APPROVED** [review]
LGTM. Thanks.
